mypy/checker.py

```diff
@@ -287,6 +290,96 @@ class PartialTypeScope(NamedTuple):
     is_local: bool
 
 
+class InstanceDeprecatedVisitor(TypeVisitor[None]):
```

Comment:
Sorry, this is not what I meant, you should inherit `TypeTraverserVisitor` instead, and override _only_ `visit_instance()` and maybe `visit_type_alias_type()` (the latter is only needed if you want to show an error at each use place, instead of only at the type alias definition r.h.s.)

Reply:
No problem.  `TypeTraverserVisitor` is the more convenient base class, of course.  Now I know it exists.

I also changed the name `visited` to `seen_aliases` (in agreement with `TypeArgumentAnalyzer`).

I was unsure whether emitting deprecation notes due to **using** type alias definitions is a good idea.  I had a slight tendency towards it, so I implemented it this way for now.  Maybe @JelleZijlstra has a better thought-out opinion on this?